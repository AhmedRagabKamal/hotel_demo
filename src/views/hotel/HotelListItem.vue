<template>
  <div class="col-sm-4">
    <div class="card mb-3">
      <div class="card-body text-left">
        <button @click="getHotelDetail(hotel.id)"
         class="btn btn-link">{{hotel.name}}</button>
        <div class="row">
          <div class="col-sm-5">
              <img :src="hotel.photo" class="img-thumbnail" :alt="hotel.name" />
          </div>
          <div class="col-sm-7">
            <ul class="list-unstyled">
              <li>{{hotel.pricePerNight * nightsCount | currency('$')}}
                 for {{nightsCount}} night</li>
              <li>{{hotel.totalScore}}
                 <span v-grade="{grade:hotel.totalScore}"></span>
              </li>
              <li>{{hotel.totalReviews}} reviews</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import currency from '../../shared/filters/currency-filter';
import grade from '../../shared/directives/grade-directive';

export default {
  name: 'HotelListItem',
  props: {
    hotel: {
      required: true,
      type: Object,
      default: () => {},
    },
    nightsCount: {
      required: true,
      type: [Number, String],
      default: 1,
    },
  },
  methods: {
    getHotelDetail(hotelId) {
      this.$store.dispatch('hotel/getHotelDetails', hotelId);
    },
  },
  filters: {
    currency,
  },
  directives: {
    grade,
  },
};
</script>

<style scoped lang='scss'>

</style>
