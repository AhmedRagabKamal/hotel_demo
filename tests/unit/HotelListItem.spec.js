import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import HotelListItem from '@/views/hotel/HotelListItem.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Actions.vue', () => {
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      getHotelDetails: jest.fn(),
    };
    store = new Vuex.Store({
      state: {},
      actions,
    });
  });

  const build = () => {
    const wrapper = shallowMount(HotelListItem, {
      store,
      localVue,
      propsData: {
        hotel: {},
        nightsCount: 1,
      },
    });
    return {
      wrapper,
    };
  };

  it('render component', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  // it('dispatches "getHotelDetails" when button is clicked', () => {
  //   const { wrapper } = build();
  //   const button = wrapper.find('button');
  //   button.trigger('click');
  //   expect(actions.getHotelDetails).toHaveBeenCalled();
  // });
});
