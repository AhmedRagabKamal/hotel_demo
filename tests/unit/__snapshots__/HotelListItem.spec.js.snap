// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions.vue render component 1`] = `
<div class="col-sm-4">
  <div class="card" style="width: 18rem;">
    <div class="card-body text-left"><button class="btn btn-link"></button>
      <div class="row">
        <div class="col-sm-5"><img class="img-thumbnail"></div>
        <div class="col-sm-7">
          <ul class="list-unstyled">
            <li>$ NaN for 1 night</li>
            <li>
              <span>Accepted</span></li>
            <li> reviews</li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
